samples/bootloader/README.rst

```diff
@@ -188,53 +178,41 @@ When using the ``nrf52840dk_nrf52840`` board, this would produce a flash memory
 
    B0 flash memory layout with MCUboot
 
-.. note::
-
-   In devices that store provisioning data in an OTP region, the *Provision* area does not appear in the flash memory layout.
-   See :ref:`bootloader_provisioning_otp` for more information.
-
 .. _bootloader_signature_keys:
 
-Signature Keys
+Signature keys
 ==============
 
-This bootloader supports the ECDSA-p256 key type for validating the next image in the boot chain.
-
+The ECDSA-p256 key type is supported for validating the next image in the boot chain.
 
 By default, when not explicitly defined, a private/public key pair is generated during the build.
 However, these key pairs should only be used during development.
 See :ref:`ug_fw_update_development_keys` for more details.
 
-For details on creating and using custom signature keys with this sample, refer to the :ref:`ug_bootloader_adding_immutable_keys` documentation.
+For details on creating and using custom signature keys, refer to the :ref:`ug_bootloader_adding_immutable_keys` documentation.
 
 .. _bootloader_monotonic_counter:
 
-Monotonic Counter
+Monotonic counter
 =================
 
 .. bootloader_monotonic_counter_start
 
-Firmware versions using the |NSIB| are kept in the form of a *monotonic counter*, a hardware-based version that prevents booting an image with a lower counter value.
-Counter values are kept as slots in the flash memory, with each new counter value occupying a new slot.
-See :kconfig:option:`CONFIG_SB_MONOTONIC_COUNTER` for more details.
+A non-volatile *monotonic counter* can be stored in the *Provision* area and is used to implement anti-rollback protection.
 
-You can set the counter value for different elements in the bootloader sequence.
-Using :kconfig:option:`CONFIG_FW_INFO_FIRMWARE_VERSION` option in ``prj.conf`` will allow you to change the version of the application image.
-However, in order to specify :kconfig:option:`CONFIG_FW_INFO_FIRMWARE_VERSION` for the MCUboot image do the following:
+Counter updates are written to slots in the *Provision* area, with each new counter update occupying a new slot.
+For this reason, the number of counter updates, and therefore firmware version updates, is limited.
+Enable the counter with the :kconfig:option:`CONFIG_SB_MONOTONIC_COUNTER` option.
 
-1. Create a ``child_image`` folder in the project directory.
-#. In the ``child_image`` directory create a file named ``mcuboot.conf``
-#. Add the :kconfig:option:`CONFIG_FW_INFO_FIRMWARE_VERSION` option to this file.
+The supported number of updates depends on the size of the *Provision* area and how much of that area is taken up by other features, like public key hashes.
+See option :kconfig:option:`CONFIG_SB_NUM_VER_COUNTER_SLOTS` for more details.
 
-For more information regarding the child image configuration refer to :ref:`ug_multi_image_variables` section.
+To set the counter value for an image, build it using the :kconfig:option:`CONFIG_FW_INFO_FIRMWARE_VERSION` option set appropriately:
```

Comment:
Why has the information on how to set `CONFIG_FW_INFO_FIRMWARE_VERSION` for a child image (mcuboot) been removed? This was recently added in another PR.

Reply:
Configuring child images is already documented in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_multi_image.html#image-specific-variables.

Repeating this here increases maintenance, has a risk of getting outdated, and has a risk of being lower quality, and missing details, as the common documentation for configuring child images.

Reply:
Wouldn't all projects using the immutable bootloader needs to set `CONFIG_FW_INFO_FIRMWARE_VERSION` for the second stage bootloader if they actually want to update it? The point of the immutable bootloader is to be able to update the second stage bootloader, right? So why shouldn't the sample explain/show exactly how to do that?

For a real production application it would make sense to put the version number in version control. If not the developer have to guess what version of mcuboot is running on the devices in production, and that can be hard to figure out if you haven't thought of it already. That's why I think the child_image solution is the only sensible solution for this exact problem, and I think it would be nice to tell this to the customers instead of hiding it deep in a section on another page. Yes, it's a bit more maintenance cost for us, but hopefully saves the customer for a lot of frustration.

At a minimum, we should explain that `CONFIG_FW_INFO_FIRMWARE_VERSION` needs to be set for the image of the second stage bootloader (usually mcuboot) and link to the multi image page.

Reply:
@annwoj FYI

Reply:
"The point of the immutable bootloader is to be able to update the second stage bootloader, right?"
Not entirely right. Note that NSIB can boot any image that implements FW_INFO. So you could have NSIB->NCS_application. It's not just NSIB->MCUBoot->NCS_application that is supported.

"Wouldn't all projects using the immutable bootloader needs to set CONFIG_FW_INFO_FIRMWARE_VERSION for the second stage bootloader if they actually want to update it? "
No, it defaults to 1, so they can update MCUBoot after first deploying it with the default version number.

"At a minimum, we should explain that CONFIG_FW_INFO_FIRMWARE_VERSION needs to be set for the image of the second stage bootloader (usually mcuboot) and link to the multi image page."
Done.

Reply:
@sigvartmh : Please clarify if it is supported to boot non-MCUBoot images from NSIB.

Reply:
AFAIK if you set `CONFIG_SECURE_BOOT` without enabling mcuboot you should get an application which is booted and signed by NSIB it could be that you'll have to also enable fw info

Reply:
Then I say it's supported.

Interopability between NCS components is implied unless excplicitly stated otherwise.

In this PR I am changing the wording several places to make it more clear that this is supported.